packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroup.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import { Button } from '@fluentui/react-button';
+import { extraAvatarGroupClassNames } from './useAvatarGroupStyles';
+import { getInitials } from '../../utils/getInitials';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import { Label } from '@fluentui/react-label';
+import { MoreHorizontalRegular } from '@fluentui/react-icons';
+import { PopoverSurface } from '@fluentui/react-popover';
+import { useId } from '@fluentui/react-utilities';
+import type { AvatarGroupProps, AvatarGroupState } from './AvatarGroup.types';
+
+/**
+ * Create the state required to render AvatarGroup.
+ *
+ * The returned state can be modified with hooks such as useAvatarGroupStyles_unstable,
+ * before being passed to renderAvatarGroup_unstable.
+ *
+ * @param props - props from this instance of AvatarGroup
+ * @param ref - reference to root HTMLElement of AvatarGroup
+ */
+export const useAvatarGroup_unstable = (props: AvatarGroupProps, ref: React.Ref<HTMLElement>): AvatarGroupState => {
+  const {
+    children,
+    layout = 'spread',
+    maxAvatars = 5,
+    overflowIndicator = 'number-overflowed',
+    size = 32,
+    ...rest
```

Comment:
Rather than using this `rest` object in `getNativeElementProps`, you should use the `excludedPropNames` param of `getNativeElementProps` to omit props like `size`, etc.:

```ts
getNativeElementProps('div', {
   ...props,
   children: rootChildren,
   // etc..
  },
  /* excludedPropNames: */ ['size'],
);
```

That is more resilient to future refactoring (say someone refactored this code so `size` wasn't part of this spread; that'd result in that prop being part of `rest`, and incorrectly getting onto the native element).

It's also (probably?) more performant to skip this extra copy into `rest`.